### PR TITLE
[MODULAR] Adds an entirely new map to Nova's roster: Ouroboros!

### PR DESCRIPTION
**Pull request by zydras**
## About The Pull Request

~~RINGWORM~~ OUROBOROS is a map based around the antiquated idea of "asteroid mining" from eons past. While it doesn't quite add new content immediately to the idea of mining space rocks, the station itself is situated inside a belt of asteroids that can very well be exploited early-on for a few materials. The most noticeable quirk is that each department is equally distributed between the two Z levels of the station rather than simply being different slices. It goes up and down!

## How This Contributes To The Nova Sector Roleplay Experience

Nova has always wanted to have its own maps to play with rather than depending on TG. This map aims to add new diversity for its (small) size, allowing it to be comfortable for any kinds of gameplay and roleplay. Notably:
 - Much like Blueshift, multiple small areas are dotted around the station in both public and private scenarios in other to create roleplay-inviting spaces. Things such as a little Jim Nortons café, the Medical Recovery Ward or simply the Carp Pro Shop allow players to come up with their own ideas and have a space to enact them in.
 - The station is built on the concept that communal spaces helps build roleplay as it puts players into proximity. It can sometimes be hard to roleplay with others if one simply is never seen - as such, the goal of this map was to make most departments _visible_ to all players rather than "slotted away" to never be seen.
 - The big quirk of this station is the use of multi-z with more depth than most current-roster stations use. All, and I mean _**All**_ departments are spread between the two layers in a smooth, connected manner. This also includes a few clever tricks that allow certain departments to interact with the crew despite being deep in said department. See: Warden's Panopticon.

## Proof of Testing

What about some sneak-peaks of parts of the map ingame, hm? Obviously the entire map isn't fully recorded, figure out ingame~
NOTE: these screenshots are not final, experimentation and (constructive) criticism will likely change the design!

<details>
<summary>Arrivals!</summary>
  
Screenshot (2452)

</details>
<details>
<summary>Supply!</summary>
  
Screenshot (2450)

</details>
<details>
<summary>The Bridge!</summary>
  
Screenshot (2449)

</details>
<details>
<summary>Science!</summary>
  
Screenshot (2468)

</details>
<details>
<summary>Engineering!</summary>
  
Screenshot (2445)

</details>
<details>
<summary>Security!</summary>
  
Screenshot (2444)

</details>
<details>
<summary>Medical!</summary>
  
Screenshot (2460)

</details>
<details>
<summary>Service!</summary>
  
Screenshot (2458)

</details>
<details>
<summary>And finally, Departures!</summary>
  
Screenshot (2467)

</details>
There's much more to be seen, but I only have so little time to take screenshots.

## Changelog

:cl:
add: The asteroid belt in close proximity of Lavaland has been recently fully acquired by Nanotrasen, with a brand new station type dubbed 'Ouroboros' inserted into the field!
add: New airless variants of Moon turfs
config: Ouroboros added as a 0-80 connected mid-pop map (similar threshold as VoidRaptor)
/:cl:

### CODE REVIEW

**Comment by W4rd3nn** [comment]
yipe!!!!!

**Comment by Iamgoofball** [comment]
for the love of god please make the sec posts dep guard outposts

**Comment by zydras** [comment]
> for the love of god please make the sec posts dep guard outposts, its an original map it doesn't need to be sec themed still

Good point, I'll get to renaming

**Comment by SylvetteSylph** [comment]
As a curator main, I have concerns regarding the library. It's extremely well-decorated and looks phenomenal aesthetically which I can appreciate. I also love the number of bookshelves available and makes extremely good use of the smaller space where others have failed to. However, having the bookshelves, the primary traffic of the library, be on a separate Z level with minimal reason to visit as far away as possible from the books kinda removes the curator's major means of natural roleplay coming from the job, and there isn't really any "natural" place for the curator to stand on the other Z level.

I'm not EXACTLY sure how this could be remedied given the nature of this station. I think the area where the staircase / table and chairs are would make a much more natural location for the desk & study, but that of course would come at a cost of needing to find a new spot for the stairs.

My two admittedly far more pedantic points are that the library desk should have a second windoor, one to act as an entrance and one to act as a "desk" to transfer items like most other library desks. Also, the pet bed placement in the middle of the bookshelves feels a bit awkward for navigation because it means actually using the pet bed for the pet would cause them to act as a blockade from the bookshelves.

**Comment by zydras** [comment]
<details>
<summary>Improved the Library on SylvetteSylph's requests. </summary>

Screenshot (2485)
Screenshot (2484)

</details>

File: modular_nova/master_files/code/game/turfs/open/floor/asteroid_moon.dm [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	worm_chance = 0
+
+/turf/closed/mineral/random/stationside/moon
+	baseturfs = /turf/open/misc/asteroid/moon/airless
+
+/turf/closed/mineral/random/high_chance/moon
+	baseturfs = /turf/open/misc/asteroid/moon/airless
+
+/turf/closed/mineral/random/labormineral/moon
+	baseturfs = /turf/open/misc/asteroid/moon/airless
```
Comment (Jolly-66) [inline modular_nova/master_files/code/game/turfs/open/floor/asteroid_moon.dm:13]: Mapping module please

**Review by Jolly-66 — CHANGES_REQUESTED** [review]

File: modular_nova/master_files/code/game/turfs/open/floor/asteroid_moon.dm (hunk 1 above)
Comment (zydras) [reply modular_nova/master_files/code/game/turfs/open/floor/asteroid_moon.dm]: Done and done, boss

**Comment by GoldenAlpharex** [comment]
Jolly asked me to test-merge this, however I'll still stand by the fact that I want this map renamed to something a bit better, so it doesn't sound like a parasite.

**Comment by xiangfei-chen** [comment]
New maps are always welcome, but these departments are extremely cramped looking.

**Comment by zydras** [comment]
> New maps are always welcome, but these departments are extremely cramped looking.

@xiangfei-chen This is a map meant for Lowpop to Midpop, it won't be a mapvote choice above a 80 player count. That's why it's cramped.

**Comment by MajManatee** [comment]
I was curious about the ability to turn one of the downstairs surgical theaters into a clinic room? I always liked having one, but I'm not sure how often they really get used.

**Comment by Ketrai** [comment]
As a chef main, I'd say the kitchen is pretty well thought out. However, it's a little weird to make botany and kitchen's working space on two separate floors, divided by an elevator.  It's a minor thing, but it's nice to be able to see each other work, and easier to make request that way than over an always flooded radio chat. It's aesthetically very pleasing though.

Hence I suggest, could you make half of the trays be upstairs where the chef can meet the botanist in person commonly? It also makes it easier for people to ask for stuff at the botanist's desk when they see someone is in and don't have to yell over radio or PDA. This way you can keep the nice Z level lookdown area, and multiple botanists can have space this way.

The elevator. Whilst super cool, is also super dangerous. I'd keep it, but with half the area being up, it doesn't have to be moved constantly this way. Also, add railings! Teshari might still kill themselves but at least it's something. First shift I already saw someone get crushed.

The hallway between the kitchen counter and the bar. It'd be nice to have windows instead of walls there so I can see the bartender and interact with the restaurant customers. Also, both the bartender and the chef have to pass through an area the chef will need to use constantly. Could it be made two tiles wide by cannibalizing some space from the freezer? Making the door into the freezer behind the kitchen counter would make it less cramped if you're worried about keeping 2 tiles of space for extra storage or moving animals about. It's an area that most chefs almost never use anyways.

Thank you kindly for your work on this map, it's quite nice!

**Comment by zydras** [comment]
Excellent comment! I'll see what I can change to improve the kitchen & botany.

**Review by Jolly-66 — APPROVED** [review]
I'm personally excited to see this come to fruition. Good job Zydras, and an even greater job on creating this.

**Pull request merged**